extensions/multiline.md

```diff
@@ -0,0 +1,241 @@
+| channel | `#channel` | variable |
+| message separator | ` :` | 2 |
+| crlf | `\n\r` | 2 |
+| message | `hello ` | variable |
+
+Not all parts may appear in a message, but a safe limit should take them into account and include an extra safety buffer in case of any unexpected syntax added by a server, e.g. a statusmsg prefix.
+
+With a safety buffer of 10, the total message size available before a split is needed can be calculated as follows
+
+```
+512 - 14 - 10 - <nick> - <user> - <host> - <channel>
+```
+
+Clients might wish to perform a `WHO` query on themselves when connecting to a server, and keep track of numeric `396 RPL_VISIBLEHOST` to keep their `user@host` mask in sync.
+
+Clients might alternatively prefer to use a hard coded limit. A possible worst case scenario for the variable parts might be:
+
+* nick: 20
+* user: 20
+* host: 63
+* channel: 32
+
+Which would leave 353 bytes for each message.
+
+## Server implementation considerations
+
+This section is non-normative.
+
+In the context of flood protection, keep in mind that clients might send an entire batched multiline message all at once.
+
+## Examples
+
+This section is non-normative.
+
+NOTE: In these examples, `<SPACE>` indicates a space character which would otherwise not be clearly visible.
+
+Client sending a mutliline batch
+
+    Client: BATCH +123 draft/multiline #channel
+    Client: @batch=123 PRIVMSG #channel hello
+    Client: @batch=123 privmsg #channel :how is<SPACE>
+    Client: @batch=123;draft/multiline-concat PRIVMSG #channel :everyone?
+    Client: BATCH -123
+
+Server sending the same batch
+
+    Server: @msgid=xxx;account=account :n!u@h BATCH +123 draft/multiline #channel
+    Server: @batch=123 :n!u@h PRIVMSG #channel hello
+    Server: @batch=123 :n!u@h PRIVMSG #channel :how is<SPACE>
+    Server: @batch=123;draft/multiline-concat :n!u@h PRIVMSG #channel :everyone?
+    Server: BATCH -123
+
+Server sending messages to clients without multiline support
+
+    Server: @msgid=xxx;account=account :n!u@h PRIVMSG #channel hello
+    Server: @account=account :n!u@h PRIVMSG #channel :how is<SPACE>
+    Server: @account=account :n!u@h PRIVMSG #channel :everyone?
+
+Final concatenated message
+
+    hello
+    how is everyone?
+
+---
+
+Invalid multiline batch target
+
+    Client: BATCH +456 draft/multiline #foo
+    Client: @batch=456 PRIVMSG #bar hello
+    Client: BATCH -456
+
+    Server: :irc.example.com FAIl BATCH MULTILINE_INVALID_TARGET #foo #bar :Invalid multiline target
+
+---
+
+`max-bytes` exceeded multiline batch
+    
+    Server: CAP * LS :draft/multiline=max-bytes=40000
+
+    Client: BATCH +abc draft/multiline #channel
+    Client: @batch=abc PRIVMSG #channel hello
+    ... lines that exceeed the max-bytes=40000 limit ...
+
+    Server: :irc.example.com FAIl BATCH MULTILINE_MAX_BYTES 40000 :Multiline batch max-bytes exceeded
+
+`max-bytes` exceeded multiline batch
+    
+    Server: CAP * LS :draft/multiline=max-lines=10
```

Comment:
This cap value is technically invalid, since `max-bytes` is mandatory.

Also, the title of this section still says `max-bytes`.